src/anomalib/deploy/export.py

```diff
@@ -101,255 +101,261 @@ def convert_center_crop(self) -> None:
+            This example demonstrates how to export the Lightning Model to OpenVINO IR.
+
+            >>> from anomalib.models import Patchcore
+            >>> from anomalib.data import Visa
+            ...
+            >>> datamodule = Visa()
+            >>> model = Patchcore()
+            ...
+            >>> model.to_openvino(
+            ...     export_root="path/to/export",
+            ...     transform=datamodule.test_data.transform,
+            ...     task=datamodule.test_data.task
+            ... )
+
+            Using Custom Transforms:
+            This example shows how to use a custom ``Transform`` object for the ``transform`` argument.
+
+            >>> from torchvision.transforms.v2 import Resize
+            >>> transform = Resize(224, 224)
+            ...
+            >>> model.to_openvino(
+            ...     export_root="path/to/export",
+            ...     transform=transform,
+            ...     task="segmentation",
+            ... )
 
-    return metadata
+        """
+        transform = transform or self.transform or self.configure_transforms()
+        export_root = _create_export_root(export_root, ExportType.OPENVINO)
+        inference_model = InferenceModel(model=self.model, transform=transform, disable_antialias=True)
+        self._write_metadata_to_json(export_root, task)
+        ov_model_path = export_root / "model.xml"
+        ov_args = {} if ov_args is None else ov_args
+        ov_args.update({"example_input": torch.zeros((1, 3, 1, 1)).to(self.device)})
+        if convert_model is not None and serialize is not None:
+            model = convert_model(inference_model, **ov_args)
+            serialize(model, ov_model_path)
+        else:
+            logger.exception("Could not find OpenVINO methods. Please check OpenVINO installation.")
+            raise ModuleNotFoundError
+        return ov_model_path
+
+    def get_metadata(
+        self,
+        task: TaskType | None = None,
+    ) -> dict[str, Any]:
+        """Get metadata for the exported model.
+
+        Args:
+            task (TaskType | None): Task type.
+                Defaults to None.
+
+        Returns:
+            dict[str, Any]: Metadata for the exported model.
+        """
+        data_metadata = {"task": task}
+        model_metadata = self._get_model_metadata()
+        metadata = {**data_metadata, **model_metadata}
 
+        # Convert torch tensors to python lists or values for json serialization.
+        for key, value in metadata.items():
+            if isinstance(value, torch.Tensor):
+                metadata[key] = value.numpy().tolist()
 
-def _get_model_metadata(model: AnomalyModule) -> dict[str, torch.Tensor]:
-    """Get meta data related to normalization from model.
+        return metadata
 
-    Args:
-        model (AnomalyModule): Anomaly model which contains metadata related to normalization.
+    def _get_model_metadata(self) -> dict[str, torch.Tensor]:
```

Comment:
looking at the metada-related methods here, I'm having some trouble seeing how the responsibilities are divided between `get_metadata` and `_get_model_metadata`. Is it really necessary to have both these method or could we merge them into a single method? Also, I feel `_write_metadata_to_json` is a utility function (like `_create_export_root`) and could be moved outside of the class.